uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/controller/UniProtKBViewByControllerIT.java

```diff
@@ -0,0 +1,244 @@
+package org.uniprot.api.uniprotkb.controller;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.client.AutoConfigureWebClient;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.uniprot.api.rest.download.AsyncDownloadMocks;
+import org.uniprot.api.uniprotkb.repository.DataStoreTestConfig;
+import org.uniprot.core.json.parser.taxonomy.TaxonomyJsonConfig;
+import org.uniprot.core.taxonomy.impl.TaxonomyEntryBuilder;
+import org.uniprot.store.indexer.DataStoreManager;
+import org.uniprot.store.indexer.DataStoreManager.StoreType;
+import org.uniprot.store.search.SolrCollection;
+import org.uniprot.store.search.document.Document;
+import org.uniprot.store.search.document.taxonomy.TaxonomyDocument;
+import org.uniprot.store.search.document.uniprot.UniProtDocument;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsStringIgnoringCase;
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.log;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.uniprot.store.indexer.DataStoreManager.StoreType.TAXONOMY;
+import static org.uniprot.store.indexer.DataStoreManager.StoreType.UNIPROT;
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(controllers = UniProtKBViewByController.class)
+@ContextConfiguration(classes = {DataStoreTestConfig.class, AsyncDownloadMocks.class})
+@AutoConfigureWebClient
+@ActiveProfiles("viewbytest")
+class UniProtKBViewByControllerIT {
```

Comment:
it was not possible since the offline profile has the bean named "uniProtKBSolrClient" mocked with mockito. But I had to use a more real one. So I did it with a different profile.
Is there any better way to do so?